clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/DeployAppengineAtomicOperation.groovy

```diff
@@ -40,18 +41,23 @@ class DeployAppengineAtomicOperation implements AtomicOperation<DeploymentResult
   AppengineJobExecutor jobExecutor
 
   DeployAppengineDescription description
+  boolean usesGcs
 
   DeployAppengineAtomicOperation(DeployAppengineDescription description) {
     this.description = description
+    this.usesGcs = description.repositoryUrl.startsWith("gs://")
   }
+
   /**
    * curl -X POST -H "Content-Type: application/json" -d '[ { "createServerGroup": { "application": "myapp", "stack": "stack", "freeFormDetails": "details", "repositoryUrl": "https://github.com/organization/project.git", "branch": "feature-branch", "credentials": "my-appengine-account", "configFilepaths": ["app.yaml"] } } ]' "http://localhost:7002/appengine/ops"
    * curl -X POST -H "Content-Type: application/json" -d '[ { "createServerGroup": { "application": "myapp", "stack": "stack", "freeFormDetails": "details", "repositoryUrl": "https://github.com/organization/project.git", "branch": "feature-branch", "credentials": "my-appengine-account", "configFilepaths": ["app.yaml"], "promote": true, "stopPreviousVersion": true } } ]' "http://localhost:7002/appengine/ops"
    * curl -X POST -H "Content-Type: application/json" -d '[ { "createServerGroup": { "application": "myapp", "stack": "stack", "freeFormDetails": "details", "repositoryUrl": "https://github.com/organization/project.git", "branch": "feature-branch", "credentials": "my-appengine-account", "configFilepaths": ["runtime: python27\napi_version: 1\nthreadsafe: true\nmanual_scaling:\n  instances: 5\ninbound_services:\n - warmup\nhandlers:\n - url: /.*\n   script: main.app"],} } ]' "http://localhost:7002/appengine/ops"
    */
   @Override
   DeploymentResult operate(List priorOutputs) {
-    def directoryPath = getFullDirectoryPath(description.credentials.localRepositoryDirectory, description.repositoryUrl)
+    def  baseDir = description.credentials.localRepositoryDirectory
+    def  directoryPath = getFullDirectoryPath(baseDir, description.repositoryUrl)
+    def  localRepositoryPath = cloneOrUpdateLocalRepository(directoryPath, 1)
```

Comment:
`cloneOrUpdateLocalRepository` should happen inside `atomicWrapper` below. It's important for the git repo case, though I think not for the gcs case.